src/Symfony/Component/Validator/Tests/Constraints/UniqueValidatorTest.php

```diff
@@ -99,4 +107,132 @@ public function testInvalidValueNamed()
             ->setCode(Unique::IS_NOT_UNIQUE)
             ->assertRaised();
     }
+
+    /**
+     * @dataProvider getCallback
+     */
+    public function testExpectsUniqueObjects($callback)
+    {
+        $object1 = new \stdClass();
+        $object1->name = 'Foo';
+        $object1->email = 'foo@email.com';
+
+        $object2 = new \stdClass();
+        $object2->name = 'Foo';
+        $object2->email = 'foobar@email.com';
+
+        $object3 = new \stdClass();
+        $object3->name = 'Bar';
+        $object3->email = 'foo@email.com';
+
+        $value = [$object1, $object2, $object3];
+
+        $this->validator->validate($value, new Unique([
+            'normalizer' => $callback,
+        ]));
+
+        $this->assertNoViolation();
+    }
+
+    /**
+     * @dataProvider getCallback
+     */
+    public function testExpectsNonUniqueObjects($callback)
+    {
+        $object1 = new \stdClass();
+        $object1->name = 'Foo';
+        $object1->email = 'bar@email.com';
+
+        $object2 = new \stdClass();
+        $object2->name = 'Foo';
+        $object2->email = 'foo@email.com';
+
+        $object3 = new \stdClass();
+        $object3->name = 'Foo';
+        $object3->email = 'foo@email.com';
+
+        $value = [$object1, $object2, $object3];
+
+        $this->validator->validate($value, new Unique([
+            'message' => 'myMessage',
+            'normalizer' => $callback,
+        ]));
+
+        $this->buildViolation('myMessage')
+            ->setParameter('{{ value }}', 'array')
+            ->setCode(Unique::IS_NOT_UNIQUE)
+            ->assertRaised();
+    }
+
+    public function getCallback()
+    {
+        return [
+            yield 'static function' => [static function (\stdClass $object) {
+                return [$object->name, $object->email];
+            }],
+            yield 'callable with string notation' => ['Symfony\Component\Validator\Tests\Constraints\CallableClass::execute'],
+            yield 'callable with static notation' => [[CallableClass::class, 'execute']],
+            yield 'callable with object' => [[new CallableClass(), 'execute']],
+        ];
+    }
+
+    public function testExpectsInvalidNonStrictComparison()
+    {
+        $callback = static function ($item) {
+            return (int) $item;
+        };
+
+        $this->validator->validate([1, '1', 1.0, '1.0'], new Unique([
+            'message' => 'myMessage',
+            'normalizer' => $callback,
+        ]));
```

Comment:
```suggestion
        $this->validator->validate([1, '1', 1.0, '1.0'], new Unique([
            'message' => 'myMessage',
            'normalizer' => 'intval',
        ]));
```

We could simplify those tests a bit, if you want.